int64 start_x
int64 start_y
int64 end_x
int64 end_y
---
float32[] rx
float32[] ry